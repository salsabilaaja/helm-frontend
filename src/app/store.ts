import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit'
import answearReducer from '../features/answearSlice'


export const store = configureStore({
  reducer: {
    answear: answearReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
