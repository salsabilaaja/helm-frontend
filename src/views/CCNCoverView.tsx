import React from "react"
import { Link } from 'react-router-dom'
import CONFIG from '../static/config.json'
import CCNActionButton from "../components/button/CCNActionButton"


function CCNCoverView() {
  return (
    <div className="coverContainer">
      <div className="coverContentContainer">
        <div className="coverContent">
          <p className="txtCoverTitle">Know your Helmet</p>
          <p className="txtCoverDesc mt-[10px]">Find you perfect helmet today in HELMETCO</p>

          <Link to={CONFIG.pages.endpoint.question}>
            <CCNActionButton
              className="btnStartCover mt-[20px]"
              a_strText="Start Now"
              filled
            />
          </Link>
        </div>
        <div className="coverPicture">
          <img src={require('../assets/img/helmCover.png')} alt="helm_cover" />
        </div>
      </div>
    </div>
  )
}

export default CCNCoverView
