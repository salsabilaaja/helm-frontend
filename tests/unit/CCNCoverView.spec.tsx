import React from 'react'
import { render } from "@testing-library/react"
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import { store } from '../../src/app/store'
import { Provider } from 'react-redux'

import CCNCoverView from "../../src/views/CCNCoverView"


describe("CCNCoverView.tsx", () => {
  const history = createMemoryHistory({ initialEntries: ['/'] })

  it("Render Properly", async () => {
    const wrapper = render(
      <Router location={history.location} navigator={history}>
        <Provider store={store}>
          <CCNCoverView />
        </Provider>
      </Router>
    )
    await expect(wrapper).toMatchSnapshot()
  })

  it('Routing between Pages', () => {
    const { container } = render(
      <Router location={history.location} navigator={history}>
        <Provider store={store}>
          <CCNCoverView />
        </Provider>
      </Router>
    )
    expect(history.location.pathname).toBe('/')

    const btnCheckout: HTMLElement  = container.querySelector('.btnStartCover') as HTMLElement
    btnCheckout.click()
    expect(history.location.pathname).toBe('/question')
  })
})
