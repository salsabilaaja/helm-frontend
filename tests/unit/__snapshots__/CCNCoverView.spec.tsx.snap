// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CCNCoverView.tsx Render Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="coverContainer"
      >
        <div
          class="coverContentContainer"
        >
          <div
            class="coverContent"
          >
            <p
              class="txtCoverTitle"
            >
              Know your Helmet
            </p>
            <p
              class="txtCoverDesc mt-[10px]"
            >
              Find you perfect helmet today in HELMETCO
            </p>
            <a
              href="/question"
            >
              <button
                aria-label=""
                class="btnActionContainer px-[10px]
        btnActionFilled 
        btnStartCover mt-[20px]"
                data-mui-internal-clone-element="true"
              >
                <p
                  class="btnActionText"
                >
                  Start Now
                </p>
              </button>
            </a>
          </div>
          <div
            class="coverPicture"
          >
            <img
              alt="helm_cover"
              src="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="coverContainer"
    >
      <div
        class="coverContentContainer"
      >
        <div
          class="coverContent"
        >
          <p
            class="txtCoverTitle"
          >
            Know your Helmet
          </p>
          <p
            class="txtCoverDesc mt-[10px]"
          >
            Find you perfect helmet today in HELMETCO
          </p>
          <a
            href="/question"
          >
            <button
              aria-label=""
              class="btnActionContainer px-[10px]
        btnActionFilled 
        btnStartCover mt-[20px]"
              data-mui-internal-clone-element="true"
            >
              <p
                class="btnActionText"
              >
                Start Now
              </p>
            </button>
          </a>
        </div>
        <div
          class="coverPicture"
        >
          <img
            alt="helm_cover"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
